_pytest/compat.py

```diff
@@ -82,7 +82,10 @@ def num_mock_patch_args(function):
     return len(patchings)
 
 
-def getfuncargnames(function, startindex=None):
+def getfuncargnames(function, startindex=None, cls=None):
+    if startindex is None and cls is not None:
```

Comment:
this bit should be documented as a hack and factored out when revisiting fixtures